src/provider/provider_os_memory.c

```diff
@@ -0,0 +1,299 @@
+/*
+ * Copyright (C) 2022-2023 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+*/
+
+#include <assert.h>
+#include <errno.h>
+#include <stddef.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/mman.h>
+#include <unistd.h>
+
+#include <umf.h>
+#include <umf/memory_provider_ops.h>
+#include <umf/providers/provider_os_memory.h>
+
+typedef struct os_memory_provider_s {
+    umf_os_memory_provider_params_t params;
+} os_memory_provider_t;
+
+static __thread int32_t TLS_last_native_error;
+
+static const char *Native_error_str[] = {
```

Comment:
nit: Instead of relying on the error values declaration order, you can set the strings to the corresponding error values likes so:
```
static const char *Native_error_str[] = {
    [UMF_OS_RESULT_SUCCESS] = "Success",
    [UMF_OS_RESULT_ERROR_WRONG_ALIGNMENT] = "wrong alignment (not a power of 2)",
    ...
}
```

Reply:
Done